pkg/ts/recipe/ParticleCook.ts

```diff
@@ -30,8 +30,17 @@ export class ParticleCook {
   }
   static specToMeta(spec) {
     // TODO(sjmiles): impedance mismatch here is likely to cause problems
-    const {$kind: kind, $container: container, $staticInputs: staticInputs, $bindings: bindings} = spec;
-    return {kind, staticInputs, bindings, container};
+    const {$kind: kind, $container: container, $staticInputs: staticInputs, $bindings: bindings, $inputs: inputs, $outputs: outputs} = spec;
+    this.formatBindings(inputs);
```

Comment:
I believe this ends up modifying the `spec`, is that intended?

Reply:
yes - it unifies the string vs object elements, so if spec was:
$inputs: [
  'foo', 
  bar: 'barStore'
]

the `inputs` would be: [{foo: 'foo'}, {bar: 'barStore'}] (probably i should update it to be {foo: ''} instead)
so that the consumers of the spec don't need to check type is string or object..

Reply:
Re: modifies "spec", I mean, I believe this code modifies the ACTUAL spec object that exists inside the recipe, instead of creating a new object.

Reply:
you're right! updated the code to create new objects instead.